src/__test__/ProductComponent.test.js

```diff
@@ -0,0 +1,21 @@
+import { render, screen } from '@testing-library/react';
+import { Provider } from 'react-redux';
+import ProductComponent from '../containers/ProductComponent';
+import store from '../redux/store';
+import '@testing-library/jest-dom';
+
+describe('ProductComponent', () => {
+  test('renders', () => {
+    render(<Provider store={store}><ProductComponent /></Provider>);
+    const heading = screen.getByRole('heading', {
+      name: 'search by category',
+    });
+    expect(heading).toBeInTheDocument();
+
+    const inputfield = screen.getByRole('textbox', {});
+    expect(inputfield).toBeInTheDocument();
+
+    const placeholder = screen.getByPlaceholderText('woman, man, jewerly...');
+    expect(placeholder).toBeInTheDocument();
```

Comment:
- [ ]  You should write unit tests for several components of the app and some pure functions in the Redux code. However, I couldn't find tests for pure functions from your Redux. Please implement unit tests for your Redux code as well so that your app can be thoroughly tested and to help you avoid errors in the future.